components/dada-parse/src/parser/code.rs

```diff
@@ -312,13 +312,13 @@ impl CodeParser<'_, '_> {
         } else if let Some((id_span, id)) = self.eat(Identifier) {
             tracing::debug!("identifier");
             Some(self.add(ExprData::Id(id), id_span))
```

Comment:
We want to be careful and look at the binary operator parsing; we can push that to a follow-up PR, but let's at least add tests here (with `#! FIXME`). The question is how to think about binary operators and newlines. Rust had the same issue to wrestle with and I suspect we want the same general answer. I'm referring to things like this:

```
fn foo() -> {
    if true { 1 } else { 2 }
    -5 # probably wants to return `-5`, not `-4`
}

fn foo() -> {
    a = if true { 1 } else { 2 }
    -5 # probably wants to return `-5` and set `a` to 1
}

fn foo() -> {
    a = (if true { 1 } else { 2 }
    -5) # probably wants to set `a` to `-4` and return `()`? Not sure.
}

fn foo() -> {
    a = if false { 1 } else { 2
    -5} # probably wants to set `a` to `-3`, but I'm not entirely sure ,especially since the next example...
}

async fn foo() -> {
    a = if false { 1 } else { print(2).await
    -5} # ...probably wants to print the number `2` and then set `a` to `-5`, and not try to subtract `5` from `()`
}
```

I'm somewhat tempted to just say "binary operators cannot be preceded by a newline", so that you have to write `b - \n 5` and not `b \n - 5`. That'd be a very simple rule.

Reply:
But opening a Question: issue to hash it out seems good.